### PR TITLE
Fixes for the official build.

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
For SqlClient, specify the runime package as a dependency instead of a
runtime dependency.
Port the following commits to unblock the official test build.

01c56fc2e5643068bf87e3dbce991a8a59644745
6753ec492e208638bdc2676a873e6ddc57262beb
7f051c9c057903905ee24596e4c7822f02386fcb

### CODE REVIEW

File: src/Native/pkg/runtime.native.System.Data.SqlClient.sni/runtime.native.System.Data.SqlClient.sni.pkgproj [hunk 1]
```diff
@@ -10,14 +10,14 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <RuntimeDependency Include="runtime.win7-amd64.runtime.native.System.Data.SqlClient.sni">
+    <Dependency Include="runtime.win7-amd64.runtime.native.System.Data.SqlClient.sni">
       <TargetRuntime>win7-amd64</TargetRuntime>
```
Comment (ericstj) [inline src/Native/pkg/runtime.native.System.Data.SqlClient.sni/runtime.native.System.Data.SqlClient.sni.pkgproj]: You need to remove TargetRuntime here.

Comment (ericstj) [inline src/Native/pkg/runtime.native.System.Data.SqlClient.sni/runtime.native.System.Data.SqlClient.sni.pkgproj]: After doing so, can you share with me the nuspec you get when building this project as a gist.

Comment (jhendrixMSFT) [inline src/Native/pkg/runtime.native.System.Data.SqlClient.sni/runtime.native.System.Data.SqlClient.sni.pkgproj]: Removed.

**Comment by jhendrixMSFT** [comment]
/cc @joshfree 
This contains the fixes for today's package issues in the release branch.

File: UpdateDependencies.ps1 [hunk 2]
```diff
@@ -3,37 +3,47 @@
...
-# This script updates all the project.json files with the latest CoreFX build version,
+# This script updates all the project.json files with the latest build version,
 # and then creates a Pull Request for the change.
 
 param(
     [Parameter(Mandatory=$true)][string]$GitHubUser,
     [Parameter(Mandatory=$true)][string]$GitHubEmail,
     [Parameter(Mandatory=$true)][string]$GitHubPassword,
-    [Parameter(Mandatory=$true)][string]$CoreFxVersionUrl, 
+    [Parameter(Mandatory=$true)][string]$VersionFileUrl,
```
Comment (dagood) [inline UpdateDependencies.ps1]: I'll get the versions repo subscription change ready for this param rename.

File: src/System.Data.SqlClient/tests/FunctionalTests/project.json [hunk 3]
```diff
@@ -2,7 +2,10 @@
   "dependencies": {
     "Microsoft.CSharp": "4.0.1-rc3-24201-01",
     "Microsoft.NETCore.Platforms": "1.0.1-rc3-24201-01",
+    "runtime.native.System.Data.SqlClient.sni": "4.0.0-rc3-24201-01",
```
Comment (weshaggard) [inline src/System.Data.SqlClient/tests/FunctionalTests/project.json]: This might start getting out of sync given our new package split between TFS and open so we may need to remove this dependency, it shouldn't be directly needed any longer. I'm actually removing it in master right now.

File: src/Native/pkg/runtime.native.System.Data.SqlClient.sni/runtime.native.System.Data.SqlClient.sni.pkgproj [hunk 4]
```diff
@@ -10,14 +10,12 @@
...
+    <Dependency Include="runtime.win7-amd64.runtime.native.System.Data.SqlClient.sni">
       <Version>4.0.1-$(ExternalExpectedPrerelease)</Version>
-    </RuntimeDependency>
-    <RuntimeDependency Include="runtime.win7-x86.runtime.native.System.Data.SqlClient.sni">
-      <TargetRuntime>win7-x86</TargetRuntime>
+    </Dependency>
+    <Dependency Include="runtime.win7-x86.runtime.native.System.Data.SqlClient.sni">
       <Version>4.0.1-$(ExternalExpectedPrerelease)</Version>
-    </RuntimeDependency>
+    </Dependency>
```
Comment (ianhays) [inline src/Native/pkg/runtime.native.System.Data.SqlClient.sni/runtime.native.System.Data.SqlClient.sni.pkgproj]: Why the change from RuntimeDependency -> Dependency? Do we need to do this for compression as well?

Comment (jhendrixMSFT) [inline src/Native/pkg/runtime.native.System.Data.SqlClient.sni/runtime.native.System.Data.SqlClient.sni.pkgproj]: Per @ericstj 
"These have to be Dependencies rather than RuntimeDependencies. SqlClient has to be a fat package since it isn't in the shared framework, fat packages cannot rely on runtime.json."

Comment (ericstj) [inline src/Native/pkg/runtime.native.System.Data.SqlClient.sni/runtime.native.System.Data.SqlClient.sni.pkgproj]: Compression is a split package that is part of the shared framework so it is allowed to use RuntimeDependencies/runtime.json.  RID specific packages are pre-resolved and stored in the shared framework directory.
SqlClient is not part of the shared framework and must work in a shared framework (aka portable/fat) application.  As such it cannot require a RID to resolve its package dependencies and cannot use RuntimeDependencies/runtime.json.

**Comment by joshfree** [comment]
LGTM

**Comment by jhendrixMSFT** [comment]
@dotnet-bot test Innerloop Windows_NT Release Build and Test please

**Comment by dagood** [comment]
Is that Windows_NT build hanging?
